src/test/java/de/geofabrik/railway_routing/parsers/OSMRailwayClassParserTest.java

```diff
@@ -76,6 +76,18 @@ void railwayClass() {
         way.setTag("railway", "construction");
         parser.handleWayTags(edgeId, edgeIntAccess, way, relFlags);
         assertEquals(RailwayClass.CONSTRUCTION, classEnc.getEnum(false, edgeId, edgeIntAccess));
+
+        edgeIntAccess = new ArrayEdgeIntAccess(1);
+        way = new ReaderWay(29L);
+        way.setTag("railway", "proposed");
+        parser.handleWayTags(edgeId, edgeIntAccess, way, relFlags);
+        assertEquals(RailwayClass.PROPOSED, classEnc.getEnum(false, edgeId, edgeIntAccess));
+
+        edgeIntAccess = new ArrayEdgeIntAccess(1);
```

Comment:
Are these tests that came with open rail routing? I don't find them particularly readable 😅 are they just asserting that the values exist in the enum?

Reply:
yeah, they mostly came with the project. We added some additional ones as well. The test is actually exercising the class and making expectations on the output based on the inputs.